service/mapper/mapper.go

```diff
@@ -287,19 +325,30 @@ Outer:
 			log.Debug().Msg("end of delta chain reached, stopping")
 			break Outer
 		}
+
+		// Finally, we handle any kind of unexpected error by just hard failing.
 		if err != nil {
 			return fmt.Errorf("could not feed next update: %w", err)
 		}
 
+		log = log.With().Int("num_changes", len(delta)).Logger()
+
+		// If we get to this point, we found the next delta on the execution
+		// path that we are currently on. We store the step with the trie as it
+		// was before updating, and the delta that was applied. This will allow
+		// us to rewind step by step, discarding the deltas and resolving any
+		// execution forks that might appear.
+		step := Step{
+			tree:  tree,
+			delta: delta,
+		}
 		tree, err = trie.NewTrieWithUpdatedRegisters(tree, delta.Paths(), delta.Payloads())
 		if err != nil {
 			return fmt.Errorf("could not update trie: %w", err)
 		}
+		steps.PushBack(step)
```

Comment:
Why create the step before the call to `trie.NewTrieWithUpdatedRegisters` instead of between the call and `steps.PushBack(step)`? Wouldn't it be clearer?